packages/eslint-plugin-shopify/lib/rules/prefer-class-properties.js

```diff
@@ -0,0 +1,83 @@
+module.exports = function(context) {
+  var applyAlways = ((context.options[0] || 'always') === 'always');
+
+  function isSimpleLiteralProperty(prop) {
+    return !prop.computed && isSimpleLiteral(prop.value);
+  }
+
+  function isSimpleLiteral(node) {
+    return node.type === 'Literal' ||
+      (node.type === 'MemberExpression') && isSimpleLiteral(node.object) ||
+      (node.type === 'CallExpression') && isSimpleLiteral(node.callee) ||
+      (node.type === 'ArrayExpression' && node.elements.every(isSimpleLiteral)) ||
+      (node.type === 'ObjectExpression' && node.properties.every(isSimpleLiteralProperty));
+  }
+
+  function isStaticMemberExpression(node) {
+    while (node && node.type === 'MemberExpression') {
+      if (node.computed && node.property.type !== 'Literal') {
+        return false;
+      }
+
+      node = node.object;
+    }
+
+    return true;
+  }
+
+  function checkConstructorThisAssignment(node) {
+    if (isSimpleLiteral(node.right) && isStaticMemberExpression(node.left)) {
+      context.report({
+        node: node,
+        message: 'BAD!',
+      });
+    }
+  }
+
+  function checkClassDeclaration(node) {
+    if (applyAlways) {
```

Comment:
Maybe I just need some coffee beans, but I'd much prefer this to be calling methods with meaningful names.

```
checkClassDeclaration(node)
   if (applyAlways)
    var constructor = findConstructor(node)
   findMemberAssignments(constructor).forEach(... check ...)
  else
    findClassProperties(node).forEach(... BAD ...)
```

Even this has two distinct responsibilities, but I don't have to sift through all of those `===`s to get to the important stuff.